receiver/datadogrumreceiver/receiver.go

```diff
@@ -0,0 +1,164 @@
+package datadogrumreceiver
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/rs/cors"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componentstatus"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/receiverhelper"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogrumreceiver/internal/translator"
+)
+
+type datadogRUMReceiver struct {
+	address string
+	config  *Config
+	params  receiver.Settings
+
+	nextTracesConsumer consumer.Traces
+	nextLogsConsumer   consumer.Logs
+
+	server    *http.Server
+	lReceiver *receiverhelper.ObsReport
+}
+
+func newDataDogRUMReceiver(config *Config, params receiver.Settings) (component.Component, error) {
+	instance, err := receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{LongLivedCtx: false, ReceiverID: params.ID, Transport: "http", ReceiverCreateSettings: params})
+	if err != nil {
+		return nil, err
+	}
+
+	return &datadogRUMReceiver{
+		params: params,
+		config: config,
+		server: &http.Server{
+			ReadTimeout: config.ReadTimeout,
+		},
+		lReceiver: instance,
+	}, nil
+}
+
+func (ddr *datadogRUMReceiver) Start(ctx context.Context, host component.Host) error {
+	ddmux := http.NewServeMux()
+
+	ddmux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	if ddr.nextTracesConsumer != nil || ddr.nextLogsConsumer != nil {
+		ddmux.HandleFunc("/api/v2/rum", ddr.handleEvent)
+	}
+
+	var err error
+
+	// TODO: double check this
+	corsHandler := cors.New(cors.Options{
+		AllowedOrigins:   []string{"https://localhost:*", "http://localhost:*"},    // Specify allowed origins
+		AllowedMethods:   []string{"GET", "POST", "OPTIONS"},                       // Specify allowed methods
+		AllowedHeaders:   []string{"Content-Type", "Authorization", "Traceparent"}, // Specify allowed headers
+		AllowCredentials: true,                                                     // Allow credentials
+	}).Handler(ddmux)
+
+	ddr.server, err = ddr.config.ServerConfig.ToServer(
```

Comment:
```suggestion
	ddr.server, err = ddr.config.ToServer(
```
since you have this squashed and unnamed, the sub-attributes of ServerConfig should be accessible directly. Obviously if linter complains don't make this change